### ISSUE TITLE
Trouble with fine-tuning sad and scd models

### ISSUE THREAD

**Issue by nateanl** [pyannote/pyannote-audio#324]
Hi, thanks for sharing the tools! I was trying to find-tune the sad, emb, and scd models pretrained on AMI corpus. The fine-tuning of the embedding model works well. However, I met the same error when fine-tuning the sad and scd models. Here is the error information:
```
Traceback (most recent call last):
  File "/home/miniconda3/envs/dia/bin/pyannote-audio", line 8, in <module>
    sys.exit(main())
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/applications/pyannote_audio.py", line 363, in main
    app.train(protocol, **params)
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/applications/base.py", line 191, in train
    alignment=self.model_alignment_)
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/labeling/tasks/speaker_change_detection.py", line 234, in get_batch_generator
    per_epoch=self.per_epoch)
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/labeling/tasks/speaker_change_detection.py", line 99, in __init__
    resolution=resolution, alignment=alignment, **kwargs)
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/labeling/tasks/base.py", line 122, in __init__
    self.feature_extraction = Pre___ed(feature_extraction)
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/utils/path.py", line 130, in __init__
    elif Path(placeholder).is_dir():
  File "/home/miniconda3/envs/dia/lib/python3.6/pathlib.py", line 1001, in __new__
    self = cls._from_parts(args, init=False)
  File "/home/miniconda3/envs/dia/lib/python3.6/pathlib.py", line 656, in _from_parts
    drv, root, parts = self._parse_args(args)
  File "/home/miniconda3/envs/dia/lib/python3.6/pathlib.py", line 640, in _parse_args
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not RawAudio
```

Does anyone know how to solve it? Thanks!

**Comment by hbredin** [comment]
Thanks for reporting the issue.

I just pushed [a commit](https://github.com/pyannote/pyannote-audio/commit/cdfe1dfb80a4ba94085210e3bf761cc8cb74a628) that should fix the issue. 

Can you please try again with the latest version of the `develop` branch and let me know if the problem is solved?

**Comment by nateanl** [comment]
Sure. I tried to reinstall the tool and got a new error:
```
Traceback (most recent call last):
  File "/home/miniconda3/envs/dia/bin/pyannote-audio", line 8, in <module>
    sys.exit(main())
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/applications/pyannote_audio.py", line 363, in main
    app.train(protocol, **params)
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/applications/base.py", line 191, in train
    alignment=self.model_alignment_)
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/labeling/tasks/speech_activity_detection.py", line 142, in get_batch_generator
    batch_size=self.batch_size)
  File "/home/miniconda3/envs/dia/lib/python3.6/site-packages/pyannote/audio/labeling/tasks/base.py", line 122, in __init__
    self.feature_extraction = Pre___ed(feature_extraction)
NameError: name 'Pre___ed' is not defined
```

**Comment by hbredin** [comment]
Oooops. I forgot to change that. This should be fixed now.

**Comment by nateanl** [comment]
It's fixed, thanks! Another relative question, after finishing the training, can I apply the model directly on the test data? I found the tutorial requires a validation step.

**Comment by hbredin** [comment]
> It's fixed, thanks! Another relative question, after finishing the training, can I apply the model directly on the test data? I found the tutorial requires a validation step.
> For example, can I change the pretrained option to my model in raw score extraction [here](https://github.com/pyannote/pyannote-audio/tree/develop/tutorials/pipelines/speaker_diarization)?

Yes. See what the output of `pyannote-audio --help` tells you about this `--pretrained` option:
https://github.com/pyannote/pyannote-audio/blob/58f05408b4fd95c2347fc662d00f08b9867eea55/pyannote/audio/applications/pyannote_audio.py#L225-L230

https://github.com/pyannote/pyannote-audio/blob/58f05408b4fd95c2347fc662d00f08b9867eea55/pyannote/audio/applications/pyannote_audio.py#L169-L173

That being said, I recommend you first validate the model as it will help you prevent overfitting on the training set.

**Comment by hbredin** [comment]
Closing this issue as the original question has been answered.
Please open a new issue if needed.

**Comment by hbredin** [comment]
FYI, using `... apply --pretrained=/path/to/train/.../weights/xxxx.pt` was actually not supported (see #328). This should be fixed since b494803.
